cluster-autoscaler/processors/podinjection/scale_up_status_processor.go

```diff
@@ -0,0 +1,128 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podinjection
+
+import (
+	"strings"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ca_context "k8s.io/autoscaler/cluster-autoscaler/context"
+	podinjectionbackoff "k8s.io/autoscaler/cluster-autoscaler/processors/podinjection/backoff"
+	"k8s.io/autoscaler/cluster-autoscaler/processors/status"
+	"k8s.io/klog/v2"
+)
+
+// FakePodsScaleUpStatusProcessor is a ScaleUpStatusProcessor used for filtering out fake pods from scaleup status.
+type FakePodsScaleUpStatusProcessor struct {
+	fakePodControllerBackoffRegistry *podinjectionbackoff.ControllerRegistry
+}
+
+// NewFakePodsScaleUpStatusProcessor return an instance of FakePodsScaleUpStatusProcessor
+func NewFakePodsScaleUpStatusProcessor(fakePodRegistry *podinjectionbackoff.ControllerRegistry) *FakePodsScaleUpStatusProcessor {
+	return &FakePodsScaleUpStatusProcessor{
+		fakePodControllerBackoffRegistry: fakePodRegistry,
+	}
+}
+
+// Process updates scaleupStatus to remove all fake pods from
+// PodsRemainUnschedulable, PodsAwaitEvaluation & PodsTriggeredScaleup
+func (a *FakePodsScaleUpStatusProcessor) Process(_ *ca_context.AutoscalingContext, scaleUpStatus *status.ScaleUpStatus) {
+	var controllersToBackoff []types.UID
+	scaleUpStatus.PodsRemainUnschedulable, controllersToBackoff = filterNoScaleUpInfos(scaleUpStatus.PodsRemainUnschedulable)
+	scaleUpStatus.PodsAwaitEvaluation = filterFakePods(scaleUpStatus.PodsAwaitEvaluation, "PodsAwaitEvaluation")
+	scaleUpStatus.PodsTriggeredScaleUp = filterFakePods(scaleUpStatus.PodsTriggeredScaleUp, "PodsTriggeredScaleUp")
+
+	for _, uid := range controllersToBackoff {
+		a.fakePodControllerBackoffRegistry.BackoffController(uid, time.Now())
+	}
+
+}
+
+// Returns pod's controller name if found
+// Otherwise, returns an empty string
+func extractUIDsAndControllerRefNames(controllerRefs map[types.UID]*v1.OwnerReference) ([]types.UID, []string) {
+	uids := make([]types.UID, len(controllerRefs))
+	controllerRefNames := make([]string, len(controllerRefs))
+	idx := 0
+	for uid, controllerRef := range controllerRefs {
+		uids[idx] = uid
+		controllerRefNames[idx] = controllerRef.Name
+		idx++
+	}
+	return uids, controllerRefNames
+}
+
+// Removes NoScaleUpInfo entries which contain a fake pod from the input list
+// Returns a list of NoScaleUpInfo which doesn't contain any fake pods
+func filterNoScaleUpInfos(noScaleUpInfos []status.NoScaleUpInfo) ([]status.NoScaleUpInfo, []types.UID) {
```

Comment:
If we can use generics here, maybe we can just have one method instead of `filterNoScaleUpInfos` & `filterFakePods`
@x13n I remember we talked about this before but I don't remember the decision :'D

Reply:
I think one more generic filtering here would be ok, though these two functions are not identical, so I'm not sure if it is enough shared code to justify introducing a shared generic. I'm fine either way, would leave it up to @abdelrahman882 to pick one.

Reply:
Both are sharing almost same logic (filtering fake pods from a list of pods or pod wrappers), I will go with one generic function to avoid having duplicated code and logging messages